darts/models/forecasting/nbeats.py

```diff
@@ -50,11 +50,11 @@ def __init__(self, target_length):
         super().__init__()
         half_minus_one = int(target_length / 2 - 1)
         cos_vectors = [
-            torch.cos(torch.arange(target_length) * 2 * np.pi * i)
+            torch.cos(torch.arange(target_length) / target_length * 2 * np.pi * i)
             for i in range(1, half_minus_one + 1)
         ]
         sin_vectors = [
-            torch.sin(torch.arange(target_length) * 2 * np.pi * i)
+            torch.sin(torch.arange(target_length) / target_length * 2 * np.pi * i)
             for i in range(1, half_minus_one + 1)
```

Comment:
Good catch!
Now, looking again at Eq (3) in the paper, I'm wondering whether we shouldn't also change the range of `i`, as it should start from 0 for the cosines and go from ceil(H/2) to ceil(H-1) for the sines.

So something like changing the range to this for both sine and cosine:
```
for i in range(0, half_minus_one)
```
for i in range(half_minus_one + 1, 
```

And adding `target_length / 2` to `i` in the sines loop.

Unless I'm missing something?